### PR TITLE
refactor SE

### PR DESCRIPTION
**Pull request by SaitejaUtpala**

### CODE REVIEW

**Comment by SaitejaUtpala** [comment]
#1409 #1369  are blocking on this

File: geomstats/geometry/special_euclidean.py [hunk 1]
```diff
@@ -370,7 +370,7 @@ def get_point_type_shape(self, point_type=None):
         """Get the shape of the instance given the default_point_style."""
         return self.get_identity(point_type).shape
 
-    def belongs(self, point):
+    def belongs(self, point, atol=ATOL):
```
Comment (ninamiolane) [inline geomstats/geometry/special_euclidean.py:373]: gs.atol?

File: tests/tests_geomstats/test_special_euclidean.py [hunk 2]
```diff
@@ -1,330 +1,1230 @@
...
+        with expected:
+            self.metric(group)
 
-    def test_right_exp_coincides(self):
-        vector_group = SpecialEuclidean(n=2, point_type="vector")
-        theta = gs.pi / 2
-        initial_vec = gs.array([theta, 1.0, 1.0])
-        initial_matrix_vec = self.group.lie_algebra.matrix_representation(initial_vec)
+
+class TestInvariantMetricOnSE(
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_special_euclidean.py:662]: This should be: TestSpecialEuclideanMatrixCanonicalRightMetric to be consistent.

Also, correct typo in this file: Canonical (with one 'n').

File: tests/tests_geomstats/test_special_euclidean.py [hunk 3]
```diff
@@ -1,330 +1,1230 @@
...
-        expected = False
-        self.assertAllClose(result, expected)
+class TestSpecialEuclidean3Vectors(TestCase, metaclass=Parametrizer):
+    space = SpecialEuclidean
 
-        vec = gs.array([vec_1, vec_2])
-        expected = gs.array([True, False])
-        result = self.group.lie_algebra.belongs(vec)
-        self.assertAllClose(result, expected)
+    class TestDataSpecialEuclidean3Vectors(TestData):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_special_euclidean.py:844]: **TestData

File: tests/tests_geomstats/test_special_euclidean.py [hunk 4]
```diff
@@ -1,330 +1,1230 @@
...
+
+                expected_rot = gs.concatenate(
+                    [new_angle * point[:3] / angle, gs.zeros(3)], axis=0
+                )
+                expected_trans = gs.concatenate([gs.zeros(3), point[3:6]], axis=0)
+                expected = expected_rot + expected_trans
+                smoke_data += [dict(point=point, expected=expected)]
+            return self.generate_tests(smoke_data)
+
+    testing_data = TestDataSpecialEuclidean3Vectors()
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_special_euclidean.py:1129]: **TestData

**Review by ninamiolane — APPROVED** [review]
LGTM! Just minor comments inline and:
- what is the file result.png? it seems that it should not be part of this PR.

**Comment by SaitejaUtpala** [comment]
> LGTM! Just minor comments inline and:
> 
>     * what is the file result.png? it seems that it should not be part of this PR.
@ninamiolane  sorry! pytest generates this when run locally. deleted

**Comment by SaitejaUtpala** [comment]
@ninamiolane Thanks for your review!  addressed your review comments.

**Pull request merged**